src/Plugin/Field/FieldWidget/TideSiteRestrictionFieldWidget.php

```diff
@@ -141,8 +142,8 @@ protected function userOptionsFilter(AccountProxyInterface $account, array $opti
   public function massageFormValues(array $values, array $form, FormStateInterface $form_state) {
     $results = parent::massageFormValues($values, $form, $form_state);
     $options = [];
-    $entity = $form_state->getFormObject()->getEntity();
-    if (!$this->helper->canBypassRestriction($this->currentUser)) {
+    if (!$this->helper->canBypassRestriction($this->currentUser) && $form_state->getFormObject() instanceof EntityFormInterface) {
```

Comment:
I'm just trying to understand the logic here, why should this statement only run if the user can't bypass the site restriction?

Reply:
Hi @GROwen 

if the users were `site_admin` or `administrator` the `canBypassRestriction` method will return `true`. as a result, it will go to line 162. and bypass line146 to line 160.

if the user were not `site_admin` or `administrator`, and `$form_state->getFormObject() instanceof EntityFormInterface` it will check line 146 to line 160

Reply:
Cool, I'm not sure why I was finding that so hard to interpret.

In that case does any further processing need to occur in this function for users who can bypass the restriction? Could the $values just be returned as they are from the parent?

Reply:
It would improve the readability of this to first check if the user can bypass the restriction and return if they can, there shouldn't be any need to alter values if they can, should there?. i.e.
```
if (canBypassRestriction) {
  return
} else {
  // massage values
}

Reply:
hi @GROwen ,yes. I've changed the logic, which looks more clear now. please take a look. thanks.